METRICS.md

```diff
@@ -0,0 +1,153 @@
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+
+# Apache NiFi - MiNiFi - C++ Metrics Readme.
+
+
+This readme defines the metrics published by Apache NiFi. All options defined are located in minifi.properties.
+
+## Table of Contents
+
+- [Description](#description)
+- [Configuration](#configuration)
+- [Metrics](#metrics)
+
+## Description
+
+Apache NiFi MiNiFi C++ can communicate metrics about the agent's status, that can be a system level or component level metric.
+These metrics are exposed through the agent implemented metric publishers that can be configured in the minifi.properties.
+Aside from the publisher exposed metrics, metrics are also sent through C2 protocol of which there is more information in the
+[C2 documentation](C2.md#metrics).
+
+## Configuration
+
+To configure the a metrics publisher first we have to set which publisher class should be used:
+
+	# in minifi.properties
+
+	nifi.metrics.publisher.class=PrometheusMetricsPublisher
+
+Currently PrometheusMetricsPublisher is the only available publisher in MiNiFi C++ which publishes metrics to a Prometheus server.
+To use the publisher a port should also be configured where the metrics will be available to be scraped through:
+
+	# in minifi.properties
+
+	nifi.metrics.publisher.port=9936
```

Comment:
This port should be a property of PrometheusMetricsPublisher, not all publishers in general.

Reply:
If our goal is to support more ports for future publishers, wouldn't that be easier to use comma separated values for `nifi.metrics.publisher.port` for those publishers?

Reply:
It begs the question, which of those values belong to which publisher? What if a publisher doesn't use the network, or not through TCP/UDP? Or it periodically posts via HTTP, in which the URL already has the port?

We could declare a global port setting and have the publishers read from it if they want to, but it fundamentally belongs to the component that actually uses that port, if they need such a setting.

Reply:
Updated in 2f43e4e6818e72cc2e6cd5a0e22d4f96962db9eb